### PR TITLE
com.github.Murmele.scram

### PR DESCRIPTION
**Pull request by Murmele**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.scram-pra.scram.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+- name: scram
+  buildsystem: cmake-ninja
+  build-commands:
+  - desktop-file-install --dir=/app/share/applications cutecom.desktop
+  - install -Dm755 org.scram-pra.scram.appdata.xml /app/share/metainfo/org.scram-pra.scram.appdata.xml
+  - install -Dm755 gui/images/scram_log.svg /app/share/icons/hicolor/scalable/apps/scram.svg
+  sources:
+  - type: git
+    url: https://github.com/Martin2278/scram
+    branch: develop
```
Comment (hfiguiere) [inline org.scram-pra.scram.yml:22]: you need an actual revision. definitely not a branch

File: org.scram-pra.scram.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+rename-desktop-file: scram.desktop
+finish-args:
+- "--socket=wayland"
+- "--socket=fallback-x11"
+- "--filesystem=home"
+modules:
+- name: scram
+  buildsystem: cmake-ninja
+  build-commands:
+  - desktop-file-install --dir=/app/share/applications cutecom.desktop
```
Comment (hfiguiere) [inline org.scram-pra.scram.yml:16]: this doesn't match the `rename-desktop-file`

File: org.scram-pra.scram.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+rename-icon: scram
+rename-desktop-file: scram.desktop
+finish-args:
+- "--socket=wayland"
+- "--socket=fallback-x11"
+- "--filesystem=home"
+modules:
+- name: scram
+  buildsystem: cmake-ninja
+  build-commands:
```
Comment (hfiguiere) [inline org.scram-pra.scram.yml:15]:
```suggestion
  post-install:
```

File: org.scram-pra.scram.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  - desktop-file-install --dir=/app/share/applications cutecom.desktop
+  - install -Dm755 org.scram-pra.scram.appdata.xml /app/share/metainfo/org.scram-pra.scram.appdata.xml
+  - install -Dm755 gui/images/scram_log.svg /app/share/icons/hicolor/scalable/apps/scram.svg
+  sources:
+  - type: git
+    url: https://github.com/Martin2278/scram
```
Comment (hfiguiere) [inline org.scram-pra.scram.yml:21]: this appear to be a fork. Why not using upstream directly?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by Murmele** [comment]
@hfiguiere Sorry I am not ready yet. I will remove the WIP flag as soon as I am ready

**Comment by Murmele** [comment]
bot, build org.scram-pra.scram

**Comment by Murmele** [comment]
bot, build org.scram_pra.scram

File: org.scram_pra.scram.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  - -DCMAKE_BUILD_TYPE=Release
+  - -DINSTALL_PREFIX=/app
+  - -DBUILD_GUI=ON
+  - -DWITH_TCMALLOC=OFF
+  - -DWITH_JEMALLOC=OFF
+  - -DWITH_COVERAGE=OFF
+  - -DWITH_PROFILE=OFF
+  builddir: true
+  post-install:
+  - desktop-file-install --dir=/app/share/applications gui/scram-gui.desktop
```
Comment (hfiguiere) [inline org.scram_pra.scram.yml:29]: I this file already being installed but the build.

Also the path is relative to the builddir, hence it's not found. (build failure)

So you should just remove it.

File: org.scram_pra.scram.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  - install -Dm755 gui/scram-gui.appdata.xml /app/share/metainfo/org.scram_pra.scram.appdata.xml
+  - install -Dm755 gui/images/scram_log.svg /app/share/icons/hicolor/scalable/apps/scram.svg
+  - install -Dm755 gui/images/scram16x16.png /app/share/icons/hicolor/16x16/apps/scram.png
+  - install -Dm755 gui/images/scram32x32.png /app/share/icons/hicolor/32x32/apps/scram.png
+  - install -Dm755 gui/images/scram48x48.png /app/share/icons/hicolor/48x48/apps/scram.png
+  - install -Dm755 gui/images/scram64x64.png /app/share/icons/hicolor/64x64/apps/scram.png
+  - install -Dm755 gui/images/scram128x128.png /app/share/icons/hicolor/128x128/apps/scram.png
+  - install -Dm755 gui/images/scram256x256.png /app/share/icons/hicolor/256x256/apps/scram.png
```
Comment (hfiguiere) [inline org.scram_pra.scram.yml:37]: I also see the appdata and the icons being installed by the build system so this is not needed.

File: org.scram_pra.scram.yml [hunk 7]
```diff
@@ -0,0 +1,42 @@
+app-id: org.scram_pra.scram
+runtime: org.kde.Sdk
+runtime-version: 5.15-21.08
+sdk: org.kde.Sdk
+command: scram-gui
+rename-icon: scram
+rename-desktop-file: scram-gui.desktop
```
Comment (hfiguiere) [inline org.scram_pra.scram.yml:7]: You also need `rename-appdata-file: scram-gui.appdata.xml`

File: org.scram_pra.scram.yml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+- "--share=ipc"
+# - "--device=dri" # TODO: needed?
+modules:
+- boost.json
+# libxml2 is already in the sdk
+- name: scram
+  buildsystem: cmake-ninja
+  config-opts:
+  - -DCMAKE_BUILD_TYPE=Release
+  - -DINSTALL_PREFIX=/app
```
Comment (hfiguiere) [inline org.scram_pra.scram.yml:21]: you don't need this, it's done automatically.

File: org.scram_pra.scram.yml [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  - install -Dm755 gui/images/scram16x16.png /app/share/icons/hicolor/16x16/apps/scram.png
+  - install -Dm755 gui/images/scram32x32.png /app/share/icons/hicolor/32x32/apps/scram.png
+  - install -Dm755 gui/images/scram48x48.png /app/share/icons/hicolor/48x48/apps/scram.png
+  - install -Dm755 gui/images/scram64x64.png /app/share/icons/hicolor/64x64/apps/scram.png
+  - install -Dm755 gui/images/scram128x128.png /app/share/icons/hicolor/128x128/apps/scram.png
+  - install -Dm755 gui/images/scram256x256.png /app/share/icons/hicolor/256x256/apps/scram.png
+  sources:
+  - type: git
+    url: https://github.com/Murmele/scram
+    branch: develop
```
Comment (hfiguiere) [inline org.scram_pra.scram.yml:41]: off course you'll need a revision here and not a branch

**Comment by Murmele** [comment]
bot, build org.scram_pra.scram

**Comment by Murmele** [comment]
@hfiguiere In the binary folder there is also the cli application. How can I make this application also available?

For example I would run the gui with flatpak run org.scram_pra.scram-gui
and the CLI with flatpak run org.scram_pra.scram

**Comment by barthalion** [comment]
> Is there a way to rename also the id in the appdata.xml not only the filename?

This has to be done manually.

> For example I would run the gui with flatpak run org.scram_pra.scram-gui and the CLI with flatpak run org.scram_pra.scram

There's no way to expose multiple applications that way. Users can run `flatpak run --command=cliprogramname org.scram_pra.scram`.

File: org.scram-pra.scram.yml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+finish-args:
+- "--socket=wayland"
+- "--socket=fallback-x11"
+- "--filesystem=home"
+modules:
+- name: scram
+  buildsystem: cmake-ninja
+  build-commands:
+  - desktop-file-install --dir=/app/share/applications cutecom.desktop
+  - install -Dm755 org.scram-pra.scram.appdata.xml /app/share/metainfo/org.scram-pra.scram.appdata.xml
```
[the remark this answers is not in the document]
Comment (Murmele) [reply org.scram-pra.scram.yml]: changed

File: org.scram-pra.scram.yml (hunk 4 above)
Comment (Murmele) [reply org.scram-pra.scram.yml]: the creator of the project does not react on any messages. I changed also some things and I will do it also in the future

File: org.scram_pra.scram.yml (hunk 6 above)
Comment (Murmele) [reply org.scram_pra.scram.yml]: done

File: org.scram_pra.scram.yml (hunk 7 above)
Comment (Murmele) [reply org.scram_pra.scram.yml]: done

File: org.scram_pra.scram.yml (hunk 8 above)
Comment (Murmele) [reply org.scram_pra.scram.yml]: done

File: org.scram_pra.scram.yml (hunk 9 above)
Comment (Murmele) [reply org.scram_pra.scram.yml]: done

**Comment by Murmele** [comment]
bot, build com.github.Murmele.scram

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/com.github.Murmele.scram.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://screenshots.debian.net/screenshots/000/015/701/large.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://screenshots.debian.net/screenshots/000/015/589/large.png]
• tag-missing           : <release> required
• tag-missing           : <name> is not present
• tag-missing           : <summary> is not present
Validation of files failed
```

File: com.github.Murmele.scram.yml [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+- "--filesystem=home"
+- "--share=ipc"
+# - "--device=dri" # TODO: needed?
+modules:
+- boost.json
+# libxml2 is already in the sdk
+- name: scram
+  buildsystem: cmake-ninja
+  config-opts:
+  - -DCMAKE_BUILD_TYPE=Release
```
Comment (hfiguiere) [inline com.github.Murmele.scram.yml:21]:
```suggestion
  - -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

File: com.github.Murmele.scram.yml [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+command: scram-gui
+rename-icon: scram
+rename-desktop-file: scram-gui.desktop
+rename-appdata-file: scram-gui.appdata.xml
+finish-args:
+- "--socket=wayland"
+- "--socket=fallback-x11"
+- "--filesystem=home"
+- "--share=ipc"
+# - "--device=dri" # TODO: needed?
```
Comment (hfiguiere) [inline com.github.Murmele.scram.yml:14]: it should tell you on the console if this is needed. Errors about GL or "drm"

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.Murmele.scram.yml (hunk 11 above)
Comment (bilelmoussaoui) [reply com.github.Murmele.scram.yml:21]: Do you actually need to set the CMAKE_BUILD_TYPE? iirc flatpak-builder sets that already

Comment (hfiguiere) [reply com.github.Murmele.scram.yml:21]: Not that I can see:

https://github.com/flatpak/flatpak-builder/blob/386976adb8ab97c612394fd12118c7c7e477dda9/src/builder-module.c#L1775

File: com.github.Murmele.scram.yml (hunk 12 above)
Comment (Murmele) [reply com.github.Murmele.scram.yml]: I removed it, because it seems that it is not neccessary

File: com.github.Murmele.scram.yml (hunk 11 above)
Comment (Murmele) [reply com.github.Murmele.scram.yml]: changed

**Comment by Murmele** [comment]
bot, build com.github.Murmele.scram

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/com.github.Murmele.scram.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://screenshots.debian.net/screenshots/000/015/701/large.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://screenshots.debian.net/screenshots/000/015/589/large.png]
• tag-missing           : <release> required
• tag-missing           : <name> is not present
• tag-missing           : <summary> is not present
Validation of files failed
```

File: com.github.Murmele.scram.yml [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+rename-icon: scram
+rename-desktop-file: scram-gui.desktop
+rename-appdata-file: scram-gui.appdata.xml
+finish-args:
+- --socket=wayland
+- --socket=fallback-x11
+- --filesystem=home
+- --share=ipc
+modules:
+- boost.json
```
Comment (hfiguiere) [inline com.github.Murmele.scram.yml:15]: having the boost module inline here is better

Comment (Murmele) [reply com.github.Murmele.scram.yml]: done

**Comment by Murmele** [comment]
bot, build com.github.Murmele.scram

**Comment by Murmele** [comment]
bot, build com.github.Murmele.scram

File: com.github.Murmele.scram.yml [hunk 14]
```diff
@@ -0,0 +1,45 @@
+app-id: com.github.Murmele.scram
+runtime: org.kde.Sdk
+runtime-version: 5.15-21.08
+sdk: org.kde.Sdk
+command: scram-gui
+rename-icon: scram
+rename-desktop-file: scram-gui.desktop
+rename-appdata-file: scram-gui.appdata.xml
+finish-args:
```
Comment (barthalion) [inline com.github.Murmele.scram.yml:9]:
```
libGL error: MESA-LOADER: failed to retrieve device information
libGL error: MESA-LOADER: failed to open amdgpu: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/amdgpu_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/GL/default/lib/dri, suffix _dri)
libGL error: failed to load driver: amdgpu
libGL error: failed to open /dev/dri/card0: No such file or directory
libGL error: failed to load driver: radeonsi
```

You probably want to add `--device=dri` here.

File: com.github.Murmele.scram.yml [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+  - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+  - -DBUILD_GUI=ON
+  - -DWITH_TCMALLOC=OFF
+  - -DWITH_JEMALLOC=OFF
+  - -DWITH_COVERAGE=OFF
+  - -DWITH_PROFILE=OFF
+  builddir: true
+  sources:
+  - type: git
+    url: https://github.com/Murmele/scram
```
Comment (barthalion) [inline com.github.Murmele.scram.yml:43]: I noticed references to the original repo at least in the About dialog. You should update the URL there to your fork.

Comment (Murmele) [reply com.github.Murmele.scram.yml:43]: ok thanks I will do

File: com.github.Murmele.scram.yml (hunk 14 above)
Comment (Murmele) [reply com.github.Murmele.scram.yml:9]: done

**Comment by Murmele** [comment]
bot, build com.github.Murmele.scram

**Comment by Murmele** [comment]
bot, build com.github.Murmele.scram

**Pull request closed without merging**